ricecooker/classes/nodes.py

```diff
@@ -677,12 +677,12 @@ def __init__(
         role=roles.LEARNER,
         license_description=None,
         copyright_holder=None,
-        grade_levels=None,
-        resource_types=None,
-        learning_activities=None,
-        accessibility_labels=None,
-        categories=None,
-        learner_needs=None,
+        grade_levels=[],
```

Comment:
I think I see the problem - which was with the original implementation, not the updates you've done here.

These kwargs need to be moved to the `TreeNode` implementation. Then, the updates you've made to the `to_dict` method, also need to be added there, as the moment it is always returning `None` for each of these, which means it is still breaking, and also that we can't set any of this metadata for topics either.